tools/deploy_partition.py

```diff
@@ -137,17 +137,18 @@ def generate_firmware_image(board: str) -> bytes:
   return pad_to(kernel, KERNEL_SIZE) + pad_to(app, APP_SIZE)
 
 
-def load_priv_key(priv_key_file: argparse.FileType) -> Any:
+def load_priv_key(priv_key_filename: str) -> Any:
   """Loads the ECDSA private key from the specified file."""
-  priv_key = get_private_key(priv_key_file.read())
-  if not isinstance(priv_key, ec.EllipticCurvePrivateKey):
-    fatal("Private key must be an Elliptic Curve one.")
-  if not isinstance(priv_key.curve, ec.SECP256R1):
-    fatal("Private key must use Secp256r1 curve.")
-  if priv_key.key_size != 256:
-    fatal("Private key must be 256 bits long.")
-  info("Private key is valid.")
-  return priv_key
+  with open(priv_key_filename, "rb") as priv_key_file:
```

Comment:
`argparse.FileType` checks that the file exists and can be opened. Such test must now be backported here before opening the file to avoid exiting by uncaught exceptions.
If you aim at reusability, why not keeping `argparse.FileType` for the argument parsing but changing `load_priv_key` to take `bytes` (i.e. the content of the file)?

Reply:
I'd prefer that this function is self-sufficient and easy to use. I don't think we will ever want to read a private key from anything but a file.
Therefore, if we want to catch these exceptions, let's catch them.